Trustroots/trustroots
# Mobile app: improved detection

Another attempt at fixing https://github.com/Trustroots/trustroots/pull/1139#issuecomment-618893929 and unblock iOS release: https://github.com/Trustroots/trustroots-expo-mobile/issues/10

#### Proposed Changes

* Instead of waiting for native mobile app wrapper send us a signal and _then_ setting `window.isNativeMobileApp`, set it already on server-side rendered template if there's `?app` query argument.

I updated the mobile app load this way in https://github.com/Trustroots/trustroots-expo-mobile/pull/15/files#diff-958f6f56cd42c904603d226cd36a57b6R78 a while back. There might be some apps out there that don't support this but I'm not overly concerned about those; they'll keep functioning anyway.

Motivation is to make easier to modify UI for mobile app wrapper, more acutely to hide "app store badges" for iOS release ([Apple requirement](https://github.com/Trustroots/trustroots-expo-mobile/issues/10#issuecomment-528238274)).

#### Testing Instructions

* Open page via `http://localhost:3000?app`
* Ensure as logged-out user that frontpage doesn't have app store badges
* Ensure as logged-in usre that http://localhost:3000/navigation doesn't have app store badges
* Ensure that otherwise via `http://localhost:3000` those are visible


<img width="922" alt="Screenshot 2020-05-05 at 23 38 46" src="https://user-images.githubusercontent.com/87168/81113890-98e81880-8f29-11ea-8388-6d8aa3cae40c.png">
<img width="1018" alt="Screenshot 2020-05-05 at 23 38 25" src="https://user-images.githubusercontent.com/87168/81113899-9a194580-8f29-11ea-8c6e-54623f259c19.png">
<img width="320" alt="Screenshot 2020-05-05 at 23 38 12" src="https://user-images.githubusercontent.com/87168/81113906-9daccc80-8f29-11ea-9501-9f17c7f5d01a.png">
<img width="311" alt="Screenshot 2020-05-05 at 23 38 05" src="https://user-images.githubusercontent.com/87168/81113907-9e456300-8f29-11ea-8452-34bf7f09432f.png">
